px-dotnet/Insight/InsightDataManager.cs

```diff
@@ -0,0 +1,265 @@
+            try
+            {
+                var clientInfo = new ClientInfo
+                {
+                    Name = SDK.ClientName,
+                    Version = SDK.Version,
+                };
+
+                var productId = GetHeaderValue(request, HEADER_X_PRODUCT_ID);
+                var businessFlow = GetHeaderValue(request, HEADER_X_INSIGHTS_BUSINESS_FLOW);
+                BusinessFlowInfo businessFlowInfo = null;
+                if (!String.IsNullOrEmpty(productId) || !String.IsNullOrEmpty(businessFlow))
+                {
+                    businessFlowInfo = new BusinessFlowInfo
+                    {
+                        Uid = productId,
+                        Name = businessFlow,
+                    };
+                }
+
+                EventInfo eventInfo = null;
+                var eventName = GetHeaderValue(request, HEADER_X_INSIGHTS_EVENT_NAME);
+                if (!String.IsNullOrEmpty(eventName))
+                {
+                    eventInfo = new EventInfo
+                    {
+                        Name = eventName,
+                    };
+                }
+
+                var protocolHttp = new ProtocolHttp
+                {
+                    RequestUrl = request.Address.ToString(),
+                    RequestMethod = request.Method,
+                    ResponseCode = (Int32)response.StatusCode,
+                    FirstByteTime = startRequest.Subtract(start).Milliseconds,
+                    LastByteTime = endRequest.Subtract(startRequest).Milliseconds,
+                };
+
+                for (int i = 0; i < request.Headers.Count; i++)
+                {
+                    var header = request.Headers.GetKey(i);
+                    if (header.Equals(HEADER_X_INSIGHTS_DATA_ID, StringComparison.InvariantCultureIgnoreCase)
+                        || header.Equals("User-Agent", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        continue;
+                    }
+
+                    protocolHttp.AddRequestHeader(header, String.Join(";", request.Headers.GetValues(i)));
+                }
+
+                for (int i = 0; i < response.Headers.Count; i++)
+                {
+                    var header = response.Headers.GetKey(i);
+                    protocolHttp.AddResponseHeader(header, String.Join(";", response.Headers.GetValues(i)));
+                }
+
+                var protocolInfo = new ProtocolInfo
+                {
+                    Name = "http",
+                    ProtocolHttp = protocolHttp,
+                    RetryCount = retries,
+                };
+
+                var tcpInfo = new TcpInfo
+                {
+                    SourceAddress = GetHostAddress(),
+                };
+
+                var connectionInfo = new ConnectionInfo
+                {
+                    ProtocolInfo = protocolInfo,
+                    TcpInfo = tcpInfo,
+                    IsDataComplete = true,
```

Comment:
Asi lo calculan en el sdk de android, deberiamos buscar una forma similar de calcularlo con .net

    boolean isCompleteData() {
        return (this.callEndMillis > 0 || this.connectionReleasedMillis > 0 || this.responseBodyEndMillis > 0);
    }

Reply:
Cambié